File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] nums) {
        // Stack<Integer> m = new Stack<>();
        int r= nums.length-1;
        int l = r-1;
        int maxi = 0;
        while(l>=0){
            if (nums[l]>nums[r]) {
                r = l;
                l-=1;
                continue;
            }
            maxi = Math.max(maxi,nums[r]-nums[l]);
            l-=1;
        }
        return maxi;
}
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, p):
        """
        :type prices: List[int]
        :rtype: int
        """
        small = p[0]
        summ = 0
        for i in p:
            summ = max(summ,i-small)
            small = min(small,i)

        return summ 